docs/plugins/sandbox/gatsby/on-source-nodes.js

```diff
@@ -0,0 +1,61 @@
+const globby = require("globby");
+const fs = require("fs-extra");
+const path = require("path");
+const { format } = require("prettier");
+const parserTypeScript = require("prettier/parser-typescript");
+
+const { getScope, omitTypes, getByName } = require("./helpers");
+
+const parseKnobs = str => JSON.parse(JSON.parse(JSON.stringify(str.replace(/\r?\n|\r|\s|/gm, ""))));
+
+module.exports = async (
+  { actions, createNodeId, createContentDigest, reporter },
+  { path: folder },
+) => {
+  try {
+    const { createNode } = actions;
+    const files = await globby(`${folder}/**/*.tsx`);
+
+    files.forEach(file => {
+      const { name } = path.parse(file);
+
+      const exampleFolder = path.dirname(file).split("/").slice(-1).join("").toLowerCase();
+      const content = fs.readFileSync(file, "utf-8");
+      const id = [exampleFolder, "-", name.toLowerCase()].join("");
+      const scope = getScope(content);
+      const example = getByName(content, "Example");
+      const knobCode = getByName(content, "exampleKnobs");
```

Comment:
Hmm, it's almost the same as a pick with regexp, just more comfortable. I can rewrite in a single function to pick all values, not sure if it will boost the speed.

Reply:
Parsing code string to AST is much slower than couple of regexes, I think it would be really good to minimize doing it because it works on such a large scale. `getByName` can take AST instead of code string, and you can export `getCode` so that you can compute AST only once and pass it around.